Sources/SwiftParser/StringLiterals.swift

```diff
@@ -593,6 +593,37 @@ extension Parser {
       )
     }
   }
+
+  mutating func parseSimpleString() -> RawSimpleStringLiteralExprSyntax {
+    let openDelimiter = self.consume(if: .rawStringDelimiter)
+    let (unexpectedBeforeOpenQuote, openQuote) = self.expect(anyIn: SimpleStringLiteralExprSyntax.OpenQuoteOptions.self, default: .stringQuote)
+
+    /// Parse segments.
+    var segments: [RawStringSegmentSyntax] = []
+    var loopProgress = LoopProgressCondition()
+    while loopProgress.evaluate(self.currentToken) {
+      // If we encounter a token with leading trivia, we're no longer in the
+      // string literal.
+      guard currentToken.leadingTriviaText.isEmpty else { break }
+
+      if let stringSegment = self.consume(if: .stringSegment, TokenSpec(.identifier, remapping: .stringSegment)) {
+        segments.append(RawStringSegmentSyntax(content: stringSegment, arena: self.arena))
+      }
+    }
+
+    let (unexpectedBetweenSegmentAndCloseQuote, closeQuote) = self.expect(anyIn: SimpleStringLiteralExprSyntax.OpenQuoteOptions.self, default: .stringQuote)
```

Comment:
It doesn’t make a difference, but I think this should be `CloseQuoteOptions`. Also, we should default to `openDelimiter.tokenKind` so that a multi-line string literal without a closing quote produces a diagnostic that adds `"""`.

```suggestion
    let (unexpectedBetweenSegmentAndCloseQuote, closeQuote) = self.expect(anyIn: SimpleStringLiteralExprSyntax.CloseQuoteOptions.self, default: openDelimiter.tokenKind)
```